src/main/java/com/yahoo/bullet/typesystem/TypedObject.java

```diff
@@ -127,9 +135,149 @@ public int compareTo(TypedObject o) {
         }
     }
 
+    /**
+     * Get the size of the value. Currently only LIST, MAP ans STRING are supported.
```

Comment:
Typo here. Also, would it be inappropriate to let sizeOf return 1 for other types of non-null value? (or do something else besides throwing an exception)